### PR TITLE
[chore]: deprecate and remove BuildProcessorMetricName

### PR DESCRIPTION
**Pull request by VihasMakwana**
This helper is no longer needed since introduction of `mdatagen`.

Relates https://github.com/open-telemetry/opentelemetry-collector/issues/11302

### CODE REVIEW

File: processor/processorhelper/obsreport.go [hunk 1]
```diff
@@ -17,20 +16,6 @@ import (
 	"go.opentelemetry.io/collector/processor/processorhelper/internal/metadata"
 )
 
-// BuildCustomMetricName is used to be build a metric name following
-// the standards used in the Collector. The configType should be the same
-// value used to identify the type on the config.
-func BuildCustomMetricName(configType, metric string) string {
```
[the remark this answers is not in the document]
Comment (VihasMakwana) [reply processor/processorhelper/obsreport.go]: Thanks for this! 
I'll raise another PR that'll remove this, once 0.111.0 is released.

**Review by bogdandrutu — APPROVED** [review]
Please add a chlog with this.

**Comment by VihasMakwana** [comment]
@bogdandrutu Ci looks good.

**Comment by olamideTiana** [comment]
@bogdandrutu Can I work on this?

**Pull request merged**